### DISCUSSION TITLE
Traces left behind by ksu module after install/uninstall

### DISCUSSION THREAD

**Discussion by NeverSm1le** [tiann/KernelSU#1169] [Q&A]
May I ask what ksu do when install/uninstall modules, because my banking app somehow pick up this "trace".
It works with ksu no module, but when you install any mudule it will detect root and refused to run. Uninstall modules and it still refuse to run, even with clear data/reinstall. YASNAC still show passing basic intergrity so I guess it must pick up something.
Magisk get detected immediately, even with hide.
The app in question: [link](https://play.google.com/store/apps/details?id=com.vnpay.Agribank3g), support English and Vietnamese. Just put in any random 10-number phone number start with 0 (0xxxxxxxxx), if it can detect the slightest trace it will tell you the device is invalid.

[a comment by an automated account is not shown]

**Reply by NeverSm1le** [reply]
Mine is on 5.10 so I don't think that's the issue.

**Reply by NeverSm1le** [reply]
I tried 2: universal-gms-doze; viper android reborn, on seperate install (format data>flash rom+gapps>ksu>module)

**Reply by NeverSm1le** [reply]
I tried but it always show "environment is broken"

**Reply by NeverSm1le** [reply]
Gave root access to momo then turn it off made it work, here's the screenshot
![momo](https://github.com/tiann/KernelSU/assets/45039938/995f08ad-95ef-4d53-9bdd-5486693ab0fc)
Sadly still know nothing

**Reply by NeverSm1le** [reply]
Too bad it still doesn't work, guess I have to get used to no module life..

**Reply by hereblur** [reply]
I found this issue too with zygisknext.
Root Not detect before install.
After disabled or uninstall it’s still detected.
Luckily Shamika hide it for me.

[a comment by an automated account is not shown]
